@here/harp-mapview/lib/ScreenProjector.ts

```diff
@@ -49,6 +52,47 @@ export class ScreenProjector {
         return undefined;
     }
 
+    projectVectorInPlace(
+        source: THREE.Vector3 | THREE.Vector4,
+        target: THREE.Vector3
+    ): THREE.Vector3 {
+        target.x = source.x - this.m_center.x - this.m_cameraPosition.x;
```

Comment:
Keeping current code, because I assume that it saves at least a function call and allows for easiest optimization. Sometimes it just has to look like assembler if it should be fast ;)